### ISSUE TITLE
refactor time series plotting code?

### ISSUE THREAD

**Issue by iantaylor-NOAA** [r4ss/r4ss#666]
The code in `SSplotTimeseries()` is a mess. Also, better-looking time series plots can be created using `SSplotComparisons()`. 

I would love to just avoid dealing with this but someone recently reported an issue with one of the plots (uncertainty intervals shifted relative to point estimates) which I couldn't figure out the cause of in 30 minutes of debugging. I'm sure I could get there eventually, but I'm afraid that this won't be the last such problem that emerges and refactoring the code to get rid of the messy parts of `SSplotTimeseries()` seems more fun than continuing to maintain confusing code.

I could use some help deciding whether a refactor makes sense at all.
And if it does make sense, what's the best option:
1. use `SSplotComparisons()` to make time series plots even for a single model
2. pull out the timeseries plots from `SSplotComparisons()` into an independent function that is also called by `SSplotTimeseries()`
3. write entirely new time series plotting code using ggplot2 (or bring some over from another modeling platform if available)

**Comment by iantaylor-NOAA** [comment]
I finally sorted out the mismatch in years which had been caused by the switch to more general treatment of Bratio to support models that had some denominator other than B0. The range of years for Bratio in derived quantities is shorter than in the full time series (which was expected by the code even when both the point estimates and the uncertainty intervals were both from the same derived quantities). I'm not feeling inspired for a big refactor at this time, so will just close this issue and leave the `SSplotTimeseries()` code as messy as ever. 

Thanks @k-doering-NOAA and @kellijohnson-NOAA for the feedback about this.
